### PR TITLE
Internationalization support

### PR DESCRIPTION
**Pull request by LeeAlephium**
Resolves https://github.com/alephium/desktop-wallet/issues/192

~~All that remains is extracting all the translated strings.~~

### CODE REVIEW

**Comment by mvaivre** [comment]
Excellent, thanks. It's good to have this ready, so we can make sure to always add the english base string when building new components.

With that being said, we'll wait before providing any translations : 

1) We need to make sure we have the bandwidth / community involvement first to maintain up to date translations.
2) There may be some political implications depending on the language(s) we provide first. We'll wait for direct input from our community to decide which languages we wanna add.

**Comment by LeeAlephium** [comment]
There's already an issue opened to determine the language(s) to provide first: https://github.com/alephium/desktop-wallet/issues/272

File: src/pages/Wallet/WalletLayout.tsx [hunk 1]
```diff
@@ -95,52 +97,52 @@ const WalletLayout: FC = ({ children }) => {
...
       </WalletSidebar>
       <AnimatePresence exitBeforeEnter initial={true}>
         {isSendModalOpen && <SendModal onClose={() => setIsSendModalOpen(false)} />}
         {isPasswordModalOpen && (
-          <CenteredModal narrow title="Enter password" onClose={() => setIsPasswordModalOpen(false)}>
+          <CenteredModal narrow title={t`Enter password`} onClose={() => setIsPasswordModalOpen(false)}>
             <PasswordConfirmation
-              text={`Enter password for "${switchToWalletName}"`}
-              buttonText="Login"
+              text={t('Enter password for "{{switchToWalletName}}"', { switchToWalletName })}
```
Comment (nop33) [inline src/pages/Wallet/WalletLayout.tsx:144]: spaces

File: src/pages/WalletManagement/ImportWordsPage.tsx [hunk 2]
```diff
@@ -38,21 +39,24 @@ import { bip39Words } from '../../utils/bip39'
...
   const [customPlaceholder, setCustomPlaceholder] = useState(defaultPlaceholder)
   const tagifyRef = useRef<Tagify<TagData> | undefined>()
 
   const handlePhraseChange = (event: CustomEvent<ChangeEventData<BaseTagData>>) => {
     // Split words where spaces are
     const newPhrase = event.detail.value && JSON.parse(event.detail.value)
     setPhrase(newPhrase || [])
-    setCustomPlaceholder(newPhrase.length > 0 ? `${newPhrase.length} words entered` : defaultPlaceholder)
+    setCustomPlaceholder(
+      newPhrase.length > 0 ? t('{{amount}} words entered', { amount: newPhrase.length }) : defaultPlaceholder
```
Comment (nop33) [inline src/pages/WalletManagement/ImportWordsPage.tsx:58]: spaces

**Review by nop33 — COMMENTED** [review]
Hm, do we want to merge this as part of the upcoming wallet release? It won't essentially change anything until new language translations are added and we implement a language toggle, right?

/cc @LeeAlephium @mvaivre

File: src/index.tsx [hunk 3]
```diff
@@ -37,14 +38,16 @@ migrateAddressMetadata()
...
     <Router>
-      <GlobalContextProvider>
-        <AddressesContextProvider>
-          <ThemeProvider theme={lightTheme}>
-            <GlobalStyle />
-            <App />
-          </ThemeProvider>
-        </AddressesContextProvider>
-      </GlobalContextProvider>
+      <Suspense fallback="loading">
```
[the remark this answers is not in the document]
Comment (mvaivre) [reply src/index.tsx:41]: Yes. `Suspense` accepts a react node as a fallback. We could use a spinner.

File: src/contexts/global.tsx [hunk 4]
```diff
@@ -183,9 +185,13 @@ export const GlobalContextProvider: FC<{ overrideContextValue?: PartialDeep<Glob
 
   useEffect(() => {
     if (networkStatus === 'offline') {
-      setSnackbarMessage({ text: `Could not connect to the ${currentNetwork} network.`, type: 'alert', duration: 5000 })
+      setSnackbarMessage({
+        text: t('Could not connect to the {{currentNetwork}} network.', { currentNetwork }),
```
[the remark this answers is not in the document]
Comment (LeeAlephium) [reply src/contexts/global.tsx:189]: I think he means to follow code style? Sure, no problem

**Review by nop33 — APPROVED** [review]
Let's merge this after #244 is merged and we have rebased to `master`.

**Review by nop33 — APPROVED** [review]
Ready to be merged

**Comment by LeeAlephium** [comment]
:ok_hand: 

Thank you for the rebase!

**Pull request merged**